From 07166d316773164001f108f3b985a4c176d89998 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sat, 8 May 2021 23:54:46 -0400
Subject: [PATCH 1/2] sunxi: ignore mmc_auto, force SD, then eMMC

This is an opinionated change that aims to make the boot order stable,
disregarding the original boot source. The initial boot firmware should
act the same, wherever it is installed.

Additionally, this makes the automatic boot menu builder work better,
since the two boot sources are present in the boot targets.

This probably goes against the opinions and design decisions of
mainline U-Boot. And I respect their decision. It makes sense when you
consider the initial boot firmware to be part of a "product" OS image.
---
 include/configs/sunxi-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 068340aa964..49ce16c6aeb 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -257,7 +257,7 @@
 		"fi\0"
 
 #define BOOTENV_DEV_NAME_MMC_AUTO(devtypeu, devtypel, instance) \
-	"mmc_auto "
+	"mmc1 mmc0 " /* eMMC, then SD */
 
 #define BOOT_TARGET_DEVICES_MMC(func) func(MMC_AUTO, mmc_auto, na)
 #else
-- 
2.35.1


From de3e455bda62ad4f54328347ed962b4705dd7b4a Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sun, 17 Oct 2021 23:42:40 -0400
Subject: [PATCH 2/2] Revert "sunxi: Select environment MMC based on boot
 device"

This reverts commit 1011ebc72bda061cc9d34c1d6ff4506291a88657.
---
 env/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/env/Kconfig b/env/Kconfig
index 2f625b22575..0dbd16eac6b 100644
--- a/env/Kconfig
+++ b/env/Kconfig
@@ -443,7 +443,8 @@ config ENV_FAT_DEVICE_AND_PART
 	depends on ENV_IS_IN_FAT
 	default "0:1" if TI_COMMON_CMD_OPTIONS
 	default "0:auto" if ARCH_ZYNQ || ARCH_ZYNQMP || ARCH_VERSAL
-	default ":auto" if ARCH_SUNXI
+	default "0:auto" if ARCH_SUNXI && MMC_SUNXI_SLOT_EXTRA = -1
+	default "1:auto" if ARCH_SUNXI && MMC_SUNXI_SLOT_EXTRA != -1
 	default "0" if ARCH_AT91
 	help
 	  Define this to a string to specify the partition of the device. It can
-- 
2.35.1

